### PR TITLE
chore: setup PHPStan

### PR DESCRIPTION
**Pull request by justlevine**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository.

- [x] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [x] Make sure you are requesting to pull request from a **topic/feature/bugfix/devops branch** (right side). Don't pull request from your master!
- [x] Have you ensured/updated that CLI tests to extend coverage to any new logic. Learn how to modify the tests [here](https://woographql.com/contributing/2-local-testing).

What does this implement/fix? Explain your changes.
---------------------------------------------------
This PR implements PHPStan static analysis for the plugin.

Incidentally, the platform req in `composer.json` has been set to PHP 7.3 to ensure Composer deps are built against the same WordPress version.

To run (following the convention of the other composer script commands)

```bash
composer stan
```

Does this close any currently open issues?
------------------------------------------
…

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
- This intentionally does _not_ include a GH workflow, since itl just add extra noise until we address some of the smells.
- To get issues shown in VSCode's `Problems` panel, you must install the dev dependences and then reload the window.

Where has this been tested?
---------------------------

- **WooGraphQL Version:** 0.12.5
- **WPGraphQL Version:** 1.14.3
- **WordPress Version:**6.2
- **WooCommerce Version:**7.6.1

### CODE REVIEW

**Review by kidunot89 — APPROVED** [review]
@justlevine Thanks for the quick work, bro 🚀 .

File: includes/data/connection/class-product-connection-resolver.php [hunk 1]
```diff
@@ -89,7 +89,7 @@ public function get_loader_name() {
 	 */
 	public function get_node_by_id( $id ) {
 		$post = get_post( $id );
-		if ( empty( $post ) || is_wp_error( $post ) ) {
+		if ( empty( $post ) ) {
```
Comment (justlevine) [inline includes/data/connection/class-product-connection-resolver.php:92]: This is possibly a change in behavior.

File: includes/data/connection/class-product-connection-resolver.php [hunk 2]
```diff
@@ -472,7 +472,7 @@ public function sanitize_input_fields( array $where_args ) {
 				}
 
 				// Set sub tax query relation.
-				if ( 1 > count( $sub_tax_query ) ) {
+				if ( 1 > count( $sub_tax_query ) && ! empty( $relation ) ) {
```
Comment (justlevine) [inline includes/data/connection/class-product-connection-resolver.php:475]: this is possibly a change in behavior

File: includes/data/mutation/class-cart-mutation.php [hunk 3]
```diff
@@ -117,12 +117,12 @@ private static function prepare_attributes( $product_id, array $variation_data =
...
 	public static function retrieve_cart_items( $input, $context, $info, $mutation = '' ) {
+		$items = [];
 		if ( ! empty( $input['all'] ) && $input['all'] ) {
 			$items = array_values( \WC()->cart->get_cart() );
 		}
-
-		if ( ! empty( $input['keys'] ) && ! isset( $items ) ) {
-			$items = [];
+		
+		if ( ! empty( $input['keys'] ) && ! empty( $items ) ) {
```
Comment (justlevine) [inline includes/data/mutation/class-cart-mutation.php:125]: This change means that if `WC()->cart->get_cart()` returns empty, the foreach loop will no longer run.

If that's not intentional, then initialize `$items` to `null`, and then change this to `&& null !== $items`

File: includes/type/enum/class-id-type-enums.php [hunk 4]
```diff
@@ -149,6 +143,13 @@ private static function get_value( $value ) {
 					'value'       => 'uri',
 					'description' => __( 'Identify a resource by the URI.', 'wp-graphql-woocommerce' ),
 				];
+			case 'id':
+			default:
```
Comment (justlevine) [inline includes/type/enum/class-id-type-enums.php:147]: the lack of a 'default' case is a possibl change in behavior

File: includes/class-jwt-auth-schema-filters.php [hunk 5]
```diff
@@ -90,25 +90,31 @@ public static function add_jwt_output_fields( $fields, $object, $type_registry )
...
+				'type'        => 'Customer',
+				'description' => __( 'Customer object of authenticated user.', 'wp-graphql-woocommerce' ),
+				'resolve'     => function( $payload ) {
+					$id = $payload['id'];
+					return new Customer( $id );
+				},
+			]
+		);
+
+		if ( ! WooCommerce_Filters::is_session_handler_disabled() ) {
```
Comment (justlevine) [inline includes/class-jwt-auth-schema-filters.php:106]: this is a change in behavior

**Review by justlevine — COMMENTED** [review]
Okay I flagged any possible changes in behavior I noticed, but best would be to see what the GraphQL error is that we're getting from the tests

**Comment by justlevine** [comment]
@kidunot89 after my latest commit, the last two level-4 errors are:

- [includes/type/object/class-order-item-type.php#L299](https://github.com/justlevine/wp-graphql-woocommerce/blob/41712ff4064e408bea671db1262153bfd14dc3f0/includes/type/object/class-order-item-type.php#L299): `Access to an undefined property WC_Order_Item::$cached_order.`
Couldnt find core docs on this, not sure if its part of woo HPOS or a local way of passing the contextual WC_Order down WPGraphQL's resolver. If it's the latter creating a new stubfile for WC_Order_Item is the way to remediate.

- [includes/data/mutation/class-checkout-mutation.php#L512](https://github.com/justlevine/wp-graphql-woocommerce/blob/41712ff4064e408bea671db1262153bfd14dc3f0/includes/data/mutation/class-checkout-mutation.php#L512): `Method WPGraphQL\WooCommerce\Data\Mutation\Checkout_Mutation::process_order_payment() should return array but empty return statement found.`, whether you want to change the method signature so it returns null (imo more performant), or just return an empty array, you need to change [how `$results` are handled in `::process_checkout()`](https://github.com/justlevine/wp-graphql-woocommerce/blob/41712ff4064e408bea671db1262153bfd14dc3f0/includes/data/mutation/class-checkout-mutation.php#L649-L651)

**Pull request merged**